build.rs

```diff
@@ -188,6 +290,19 @@ fn load_cross_compile_info() -> Result<(InterpreterConfig, HashMap<String, Strin
     Ok((interpreter_config, fix_config_map(config_map)))
 }
 
+fn load_cross_compile_info(
+    python_include_dir: String,
+    python_lib_dir: String,
+) -> Result<(InterpreterConfig, HashMap<String, String>)> {
+    // Try to configure from the sysconfigdata file which is more accurate for the information
+    // provided at python's compile time
+    match load_cross_compile_from_sysconfigdata(&python_lib_dir) {
+        Ok(ret) => Ok(ret),
+        // If the config could not be loaded by sysconfigdata, failover to configuring from headers
+        Err(_) => load_cross_compile_from_headers(&python_include_dir, &python_lib_dir),
```

Comment:
Seems reasonable. Though I wonder if we thought the header method was broken anyway, maybe we should just remove it to simplify the build script.

Reply:
I'm not entirely sure. I think we can be fairly confident that the `sysconfigdata`  file needs to be present for the module to work properly. However, I'd like to confirm through documentation and if this is true for all versions that pyo3 support before making this removal.

Reply:
As I feared, after looking at the python source for sysconfig, [non-posix] os' don't load a `_sysconfigdata_` file, so I don't think it is possible to fully drop the header file parsing if windows is to be supported as a crosscompilation target. However, what I could do is do some conditional compilation here so all unix platforms use the `_sysconfigdata_` file and windows targets stay with the old header parsing method.

[non-posix]: https://github.com/python/cpython/blob/3.8/Lib/sysconfig.py#L426